Sources/Transaction/Memos/DestinationMemoDataUtils.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+import LibMobileCoin
+
+enum DestinationMemoUtils {
+    
+    static func isValid(
+        txOutPublicKey: RistrettoPublic,
+        txOutTargetKey: RistrettoPublic,
+        accountKey: AccountKey
+    ) -> Bool {
+        TxOutUtils.matchesSubaddress(
+             targetKey: txOutTargetKey,
+             publicKey: txOutPublicKey,
+             viewPrivateKey: accountKey.viewPrivateKey,
+             subaddressSpendPrivateKey: accountKey.changeSubaddressSpendPrivateKey)
+    }
+    
+    static func getAddressHash(
+        memoData: Data64
+    ) -> AddressHash {
+        let bytes: Data16 = memoData.asMcBuffer { memoDataPtr in
+            switch Data16.make(withMcMutableBuffer: { bufferPtr, errorPtr in
+                mc_memo_destination_memo_get_address_hash(
+                    memoDataPtr,
+                    bufferPtr,
+                    &errorPtr)
+            }) {
+            case .success(let bytes):
+                return bytes as Data16
+            case .failure(let error):
+                switch error.errorCode {
+                case .invalidInput:
+                    // Safety: This condition indicates a programming error and can only
+                    // happen if arguments to mc_memo_destination_memo_get_address_hash are
+                    // supplied incorrectly.
+                    logger.warning("error: \(redacting: error)")
+                    return Data16()
```

Comment:
the other "address hash" ffi functions always return a value because its just some computation on bytes, so in that spirit im making this always return a valid value. It makes sense because it will always succeed, and making it non-nillable simplifies a bunch of other code.